Rogue-Ape-Studios/Crusader
# ✨ Basic player animation

## Content

- A new player prefab is made, it now has a model and animations for dashing, attacking, running and idle.
- The player turns in the direction of the movement keys, but attacks in the direction of the cursor position when clicking.
- Unitask cancellation token has been added to the cooldown when attacking.

## Changes

### Added
`Assets/Animations/Player/InwardSlash.fbx` : Was created. New player animations.
`Assets/Animations/Player/OutwardSlash.fbx` : Was created. New player animations.
`Assets/Animations/Player/OverheadSlash.fbx` :Was created. New player animations.
`Assets/Animations/Player/UppercutSlash.fbx` : Was created. New player animations.
`Assets/Animations/Player/Player.controller` : Was created. New player animations controller.
`Assets/Animations/Player/PlayerDiveForward.fbx` : Was created. New player animations.
`Assets/Animations/Player/PlayerIdle.fbx` : Was created. New player animations.
`Assets/Animations/Player/RunWithSword.fbx` : Was created. New player animations.
`Assets/Prefabs/Player/Player_1.prefab` : Was created. New player prefab, with an actual model.

### Updated
`Assets/Scripts/Player/Combat/BasicAttack.cs` : Was updated. Now checks where the player clicks and turns the rigid body in that direction to attack. Also unitask.waitforseconds has now been correctly implemented.
`Assets/Scripts/Player/Combat/BasicAttack.cs` : Was updated. Animation logic has been added. The player now turns in the direction of the input instead of the cursor.

### Removed
`Assets/Prefabs/Player/BasicAttack.anim` : Was removed. This old pill animation won't be used anymore.
`Assets/Prefabs/Player/BasicAttack.anim_1` : Was removed. This old pill animation won't be used anymore.
`Assets/Prefabs/Player/BasicAttack.anim_2` : Was removed. This old pill animation won't be used anymore.
`Assets/Prefabs/Player/Idle.anim` : Was removed. This old pill animation won't be used anymore.
`Assets/Prefabs/Player/Player.controller` : Was removed. This is the old player animation controller for the pill.

## Testing

The feature / Bugfix can be testing by following these steps:

1. Drag the new player prefab in the scene.
2. Check the tags on the player's basic attack script and the environment, make sure they match.
3. You should be able to play and test the new prefab, animations (with movement direction).

Closes #64
